src/rgw/rgw_gc.cc

```diff
@@ -255,6 +260,13 @@ int RGWGC::list(int *index, string& marker, uint32_t max, bool expired_only, std
     }
 
     if (result.size() == max) {
+      if (queue_entries.size() > 0 && *truncated) {
```

Comment:
why not just check on "truncated"?

Reply:
truncated may be set to true because of entries present in omap (old style of gc) so specifically checking for queue_entries size here.